src/main/scala/wasm/AST.scala

```diff
@@ -61,12 +61,18 @@ case object Drop extends Instr
 case object Alloc extends Instr
 case object Free extends Instr
 case class Select(ty: Option[List[ValueType]]) extends Instr
-case class Block(ty: Option[ValueType], instrs: List[Instr]) extends Instr
-case class IdBlock(id: Int, ty: Option[ValueType], instrs: List[Instr]) extends Instr
-case class Loop(ty: Option[ValueType], instrs: List[Instr]) extends Instr
-case class IdLoop(id: Int, ty: Option[ValueType], instrs: List[Instr]) extends Instr
-case class If(ty: Option[ValueType], thenInstrs: List[Instr], elseInstrs: List[Instr]) extends Instr
-case class IdIf(ty: Option[ValueType], thenInstrs: IdBlock, elseInstrs: IdBlock) extends Instr
+case class Block(ty: BlockType, instrs: List[Instr]) extends Instr
+case class IdBlock(id: Int, ty: BlockType, instrs: List[Instr])
+    extends Instr
+case class Loop(ty: BlockType, instrs: List[Instr]) extends Instr
+case class IdLoop(id: Int, ty: BlockType, instrs: List[Instr])
+    extends Instr
+case class If(ty: BlockType,
```

Comment:
Maybe also increase the margin setting of the formatter?

Reply:
Got it. I’ll handle that later.